### DISCUSSION TITLE
A way to "save" an article for long term storage

### DISCUSSION THREAD

**Discussion by chuanqisun** [osmoscraft/osmosfeed#14] [Ideas]
A feature request converted from https://github.com/osmoscraft/osmosfeed/issues/6#issuecomment-831377036.

Please note, the current `cache.json` is not designed for long term storage. 
- User could wipe out the cache by mistake and any glitch in the system might cause the cache to become corrupt.
- An ever growing cache is not scalable. The UI loading will become slower as the cache gets larger.

An ideal solution would save the crawled content from the article into a permanent file separate from the cache. Searching and potentially indexing become a new problem.

This might fall out of the scope of this project but if you have other tools to recommend, please share. I can look into how osmosfeed can integrate with those tools.

Meanwhile, [osmosnote](https://github.com/osmoscraft/osmosnote) and [osmosmemo](https://github.com/osmoscraft/osmosmemo) might help preserve something you read, though neither crawls the full content.

**Comment by 2m** [comment]
I looked around and it seems a self-hosted RSS reader like miniflux (https://miniflux.app/) would be a way to go if I want my feeds to be archived.

**Reply by chuanqisun** [reply]
miniflux is awesome! I especially love the minimalist approach to make the content stand out. I'll look into the "save" mechanism and try support something similar.

**Comment by onnyyonn** [comment]
Auto-submitting the links to Internet Archive can be one easy solution. Osmosfeed can have a variable {{ArchivedLink}} that returns the Internet Archive link on the day it showed up on the RSS feed.
